remote-api/src/test/java/org/alfresco/rest/api/impl/CategoriesImplTest.java

```diff
@@ -197,5 +214,165 @@ public void testGetCategoryById_nodeNotExists()
+        then(categoryServiceMock).shouldHaveNoInteractions();
+        then(authorityServiceMock).shouldHaveNoMoreInteractions();
+    }
+
+    @Test
+    public void testCreateCategoryUnderRoot()
+    {
+        given(authorityServiceMock.hasAdminAuthority()).willReturn(true);
+        final NodeRef parentCategoryNodeRef = new NodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE, PATH_ROOT);
+        given(categoryServiceMock.getRootCategoryNodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE))
+                .willReturn(Optional.of(parentCategoryNodeRef));
+        given(nodesMock.isSubClass(parentCategoryNodeRef, ContentModel.TYPE_CATEGORY, false)).willReturn(true);
+        final NodeRef categoryNodeRef = prepareCategoryNodeRef();
+        given(categoryServiceMock.createCategory(parentCategoryNodeRef, CATEGORY_NAME)).willReturn(categoryNodeRef);
+        given(nodesMock.getNode(CATEGORY_ID)).willReturn(prepareCategoryNode());
+        final ChildAssociationRef parentAssoc = new ChildAssociationRef(null, parentCategoryNodeRef, null, categoryNodeRef);
+        given(nodeServiceMock.getPrimaryParent(categoryNodeRef)).willReturn(parentAssoc);
+        given(nodeServiceMock.getParentAssocs(parentCategoryNodeRef)).willReturn(List.of(parentAssoc));
+        given(nodeServiceMock.getChildAssocs(categoryNodeRef, RegexQNamePattern.MATCH_ALL, RegexQNamePattern.MATCH_ALL, false))
+                .willReturn(Collections.emptyList());
+
+        //when
+        final List<Category> createdCategories = objectUnderTest.createSubcategories(PATH_ROOT, prepareCategories(), parametersMock);
+
+        then(authorityServiceMock).should().hasAdminAuthority();
+        then(authorityServiceMock).shouldHaveNoMoreInteractions();
+        then(nodesMock).should().isSubClass(parentCategoryNodeRef, ContentModel.TYPE_CATEGORY, false);
+        then(nodesMock).should().getNode(CATEGORY_ID);
+        then(nodesMock).shouldHaveNoMoreInteractions();
+        then(nodeServiceMock).should().getPrimaryParent(categoryNodeRef);
+        then(nodeServiceMock).should().getParentAssocs(parentCategoryNodeRef);
+        then(nodeServiceMock).should().getChildAssocs(categoryNodeRef, RegexQNamePattern.MATCH_ALL, RegexQNamePattern.MATCH_ALL, false);
+        then(nodeServiceMock).shouldHaveNoMoreInteractions();
+        then(categoryServiceMock).should().getRootCategoryNodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE);
+        then(categoryServiceMock).should().createCategory(parentCategoryNodeRef, CATEGORY_NAME);
+        then(categoryServiceMock).shouldHaveNoMoreInteractions();
+
+        assertEquals(1, createdCategories.size());
+        Category createdCategory = createdCategories.iterator().next();
+        assertEquals(CATEGORY_ID, createdCategory.getId());
+        assertEquals(CATEGORY_NAME, createdCategory.getName());
+        assertEquals(PATH_ROOT, createdCategory.getParentId());
+        assertFalse(createdCategory.getHasChildren());
```

Comment:
Can we replace this whole block of code with an equality check?
```
List<Category> expectedCategories = List.of(Builder.something...);
assertEquals(expectedCategories, createdCategories);
```

Reply:
I replaced that with what you suggested,